.circleci/config.yml

```diff
@@ -601,31 +625,19 @@ jobs:
     parallelism: << parameters.parallel_num >>
     steps:
       - checkout_init_git
-      - browser-tools/install-browser-tools
+      - when:
+          # Deploy local UI server then run Puppeteer tests against "local ui" + "test api" environment.
+          condition: << parameters.require_local_ui >>
+          steps:
+            - skip_e2e_commit
```

Comment:
A few issues with this approach:

- Skipping on non-PR, skipping on e2e commit and skipping on lack of code change: these are all presubmit optimizations. While it's true that you also only want to run against a local UI on presubmit, it's not directly  related to local UI testing.
- There's now two parameters which overlap and could be provided inconsistently: `env_name` and `require_local_ui`

If you want to keep one job, I'd suggest renaming this param to `presubmit` or `is_presubmit`. For `start_local_ui`, I would put this in a separate conditional which is dependent only on `<<env_name>> == "local"`. Alternatively, you could avoid these issues by reverting back to two jobs, though this does result in some duplication